sos/policies/__init__.py

```diff
@@ -261,11 +265,40 @@ def get_archive_name(self):
         """
         This function should return the filename of the archive without the
         extension.
+
+        This uses the policy's name_scheme attribute to determine the name.
+        There are two pre-defined naming schemes - 'legacy' and 'friendly'
+
+        legacy - names like 'sosreport-tux.123456-20170120185433'
+        friendly - names like 'sosreport-tux-123456-2017-12-24-ezcfcop.tar.xz'
+
+        A 'custom' name_scheme can be used by a policy provided that it
+        also defines a name_scheme_string which will be formatted here,
+        using format() style string substitution. This string should also
+        include the strftime formatted substring desired to be converted.
+
+        Usable substitutions are:
+
+            name - the report name given by --name
+            case - the case id given by --case-id or --ticker-number
+            rand - a random string of 7 alpha characters
         """
-        if self.case_id:
-            self.report_name += "." + self.case_id
-        return "sosreport-%s-%s" % (self.report_name,
-                                    time.strftime("%Y%m%d%H%M%S"))
+        name = '-' + self.report_name if self.report_name != 'unused' else ''
+        case = self.case_id or ''
+        rand = ''.join(random.choice(string.lowercase) for x in range(7))
+
+        nstr = "sosreport{name}{case}"
+
+        if self.name_scheme == 'legacy':
+            case = ('.' + case if case else '')
+            nstr += '-%Y%m%d%H%M%S'
+        if self.name_scheme == 'friendly':
+            case = ('-' + case if case else '')
+            nstr += '-%Y-%m-%d-{rand}'
```

Comment:
From user experience, I prefer having also time (not only date) in the final report name. It is better for identifying and sorting reports from the same machine generated several times during the same day (well this rather happens during my testing than during a support experience).

But that is my personal point of view only.

Reply:
From #469 I dropped the time from the sosreport in favor of the random 7-character string as I believe it is easier to read at a glance than a timestamp. Even a further formatted date string is a bit harsh on the eyes in my opinion. The 7-character string can be used as both a global identifier when dealing with a mass of sosreports, and an identifier for multiple sosreports from the same day.

Reply:
Both makes sense; "harsh on the eyes" can be minimized using `-%Y-%m-%d_%H-%M-%S` format, but I understand it is subjective.

Reply:
I prefer as pmoravec++ suggested, but I'll also suggest to add hostname in the file, for cases reported against OSP, where you can get 3 controllers + several computes on the same case it will make a lot easier to differentiate them

Reply:
hostname definitely should be part of the name (how I could forget it?). There are numerous situations where one needs to work with sosreports from several systems inter-communicating and having a problem, and we should be able to distinguish among the reports easily.